### ISSUE TITLE
Pushing backspace key after the Random number: textbox is clicked on should be handled normally.

### ISSUE THREAD

**Issue by jasonkhanlar** [widelands/widelands#5527]
**Describe the bug**
In Single Player... -> New Random Game dialog

Sometimes (not always, but I'm not sure how to always reproduce) when fiddling around with changing the values of the input fields, then attempting to generate a random map results in Map Generation Error, clicking OK to close that dialog, when immediately clicking on the Random number text input field in the whitespace to the right of the numbers, pushing backspace one time (to get rid of the last number) does not get rid of the last number (again, sometimes, not always), but pushing backspace a second time then gets rid of the last number. I also was able to reproduce this without having to generate a map each time. Simply click on the checkbox for Island mode and then clicking on the Random number: textbox and push backspace once. Sometimes it works, other times nothing happens. Yesterday I was able to repeat it quite often, but repeating it again now, I can still reproduce it, but it seems less often than yesterday. I'm not sure why, but either way, something doesn't seem right. Actually, on further experimentation also, when clicking on the Random number textbox and holding the mouse button down while dragging the cursor a little bit but staying within the textbox space and then releasing the mouse cursor, then: (1) if the location of clicking and holding and dragging and releasing did not select any text, then pushing backspace first time does nothing, pushing again, it works, (2) if text was selected, then pushing backspace the first time works, no issue.

**To Reproduce**
(Read above)

**Expected behavior**
Pushing backspace key after the Random number: textbox is clicked on should be handled normally.

**Additional context**
See https://github.com/widelands/widelands/issues/5522
